contracts/protocol/bases/OfferBase.sol

```diff
@@ -180,66 +188,88 @@ contract OfferBase is ProtocolBase, IBosonOfferEvents {
+                // condition for successful payout when exchange final state is canceled
+                if (_offer.buyerCancelPenalty > offerPrice) revert InvalidOfferPenalty();
 
-            // Calculate and set the protocol fee
-            uint256 protocolFee = getProtocolFee(_offer.exchangeToken, offerPrice);
+                // Calculate and set the protocol fee
+                uint256 protocolFee = getProtocolFee(_offer.exchangeToken, offerPrice);
 
-            // Calculate the agent fee amount
-            uint256 agentFeeAmount = (agent.feePercentage * offerPrice) / 10000;
+                // Calculate the agent fee amount
+                uint256 agentFeeAmount = (agent.feePercentage * offerPrice) / 10000;
 
-            uint256 totalOfferFeeLimit = (protocolLimits().maxTotalOfferFeePercentage * offerPrice) / 10000;
+                uint256 totalOfferFeeLimit = (limits.maxTotalOfferFeePercentage * offerPrice) / 10000;
 
-            // Sum of agent fee amount and protocol fee amount should be <= offer fee limit
-            if ((agentFeeAmount + protocolFee) > totalOfferFeeLimit) revert AgentFeeAmountTooHigh();
+                // Sum of agent fee amount and protocol fee amount should be <= offer fee limit
+                if ((agentFeeAmount + protocolFee) > totalOfferFeeLimit) revert AgentFeeAmountTooHigh();
 
-            //Set offer fees props individually since calldata structs can't be copied to storage
-            offerFees.protocolFee = protocolFee;
-            offerFees.agentFee = agentFeeAmount;
+                //Set offer fees props individually since calldata structs can't be copied to storage
+                offerFees.protocolFee = protocolFee;
+                offerFees.agentFee = agentFeeAmount;
+            }
 
             // Store the agent id for the offer
-            protocolLookups().agentIdByOffer[_offer.id] = _agentId;
-        }
+            lookups.agentIdByOffer[_offer.id] = _agentId;
+
+            // Make sure that supplied royalties ok
+            // Operate in a block to avoid "stack too deep" error
+            {
+                if (_offer.royaltyInfo.recipients.length != _offer.royaltyInfo.bps.length) revert ArrayLengthMismatch();
+
+                RoyaltyRecipient[] storage royaltyRecipients = lookups.royaltyRecipientsBySeller[_offer.sellerId];
 
+                uint256 totalRoyalties;
+                for (uint256 i = 0; i < _offer.royaltyInfo.recipients.length; i++) {
+                    uint256 royaltyRecipientId = lookups.royaltyRecipientIndexBySellerAndRecipient[_offer.sellerId][
+                        _offer.royaltyInfo.recipients[i]
+                    ];
+                    if (royaltyRecipientId == 0) revert InvalidRoyaltyRecipient();
+
+                    if (_offer.royaltyInfo.bps[i] < royaltyRecipients[royaltyRecipientId - 1].minRoyaltyPercentage)
+                        revert InvalidRoyaltyPercentage();
+
+                    totalRoyalties = _offer.royaltyInfo.bps[i];
```

Comment:
shouldn't we cumulate the royalties in totalRoyalties?
```suggestion
                    totalRoyalties += _offer.royaltyInfo.bps[i];
```

Reply:
Fixed.